pkg/noderesourcetopology/pluginhelpers.go

```diff
@@ -139,12 +145,56 @@ func makePodByResourceListWithManyContainers(resources *v1.ResourceList, contain
 func extractResources(zone topologyv1alpha1.Zone) v1.ResourceList {
 	res := make(v1.ResourceList)
 	for _, resInfo := range zone.Resources {
-		klog.V(5).InfoS("Extract resources for zone", "resName", resInfo.Name, "resAvailable", resInfo.Available)
 		res[v1.ResourceName(resInfo.Name)] = resInfo.Available
 	}
 	return res
 }
 
+func logNumaNodes(desc string, nodes NUMANodeList) {
+	if !klog.V(6).Enabled() {
+		// avoid even the trivial range over nodes
+		return
+	}
+	for _, numaNode := range nodes {
+		numaIdent := fmt.Sprintf("node-%d", numaNode.NUMAID)
+		klog.V(6).InfoS(desc, resourceListToLoggable(numaIdent, numaNode.Resources)...)
+	}
+}
+
+func resourceListToLoggable(ident string, resources v1.ResourceList) []interface{} {
```

Comment:
Why the use of a slice of interface here? `map[string]string` seemed like an obvious choice to me here.

Reply:
Because doing like this enables us to write like
```
klog.V(6).InfoS("target resources", resourceListToLoggable(ident, resources)...)
```
which is both terse yet readable in the context (IMVHO) and cheap, because this way the function `resourceListToLoggable` is called (or that's the intention :) ) only when the logs need to be actually emitted, not unconditionally. Returning a map would mean we need to do some extra processing before to feed the data into `InfoS`

Reply:
Ack, this certainly helps my understanding. It looks like a trade off to me on whether we want to have a more structured data and process it before logging or have more flexibility at the time of logging and then tackle the extra processing in `kvListFormat()` when we perform the comparison. Either way, I don't feel too strongly about it and don't consider it a blocker, just want us sure that we are a conscious decision.

Reply:
In this case, it is indeed a conscious (albeit of course debatable) decision driven by the interface of `klog.InfoS`.
I'm looking for a tradeoff between readability in the logs and readability in the source code, leaning to the former when needed.
ATM I'm not aware of a clear use case for processing the return value of `resourceListToLoggable`, but if this emerges we can surely re-evaluate.